src/ucp/core/ucp_worker.c

```diff
@@ -2350,3 +2356,90 @@ void ucp_worker_print_info(ucp_worker_h worker, FILE *stream)
 
     UCP_WORKER_THREAD_CS_EXIT_CONDITIONAL(worker);
 }
+
+static int ucp_worker_keepalive_is_enabled(ucp_worker_h worker)
+{
+    return (worker->context->config.ext.keepalive_timeout != 0.0) &&
+           (worker->context->config.ext.keepalive_num_eps != 0);
+}
+
+static unsigned ucp_worker_keepalive_cb(void *arg)
+{
+    ucp_worker_h worker = (ucp_worker_h)arg;
+    double now          = ucs_get_accurate_time();
```

Comment:
why not just `ucs_get_time`?

Reply:
```ucs_get_time``` operates by seconds - not enough accuracy

Reply:
no, it operates by time units (clocks)

Reply:
ok, done